From 3f4da4ce8e760837c2a1b4f799f4b89c54bfaba9 Mon Sep 17 00:00:00 2001
From: German Cvetkov <german.cvetkov@lge.com>
Date: Wed, 30 Oct 2024 13:35:51 +0300
Subject: [PATCH] 129.0.6654.0~1: Disable SME instructions in libyuv

:Release Notes:
Disabled SME instructions in libyuv to fix webOS/OSE build after rolling
to 129.0.6654.0~1

:Detailed Notes:
In the upstream libyuv CL
https://chromium-review.googlesource.com/c/libyuv/libyuv/+/5588664
(which was rolled into Chromium in the upstream CL
https://crrev.com/c/5778746) added and used SME instructions.
It caused error on building webOS/OSE:
| cc1plus: error: invalid feature modifier 'sme' in '-march=armv9-a+sme'

Disable SME instructions in libyuv for webOS/OSE to fix this issue.

:Issues Addressed:
[NEVA-9526] Upgrade neva-main-rolling up to 6654 BUILD
---
 libyuv.gni | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/libyuv.gni b/libyuv.gni
index 3334df70..2f1bbbe2 100644
--- a/libyuv.gni
+++ b/libyuv.gni
@@ -10,6 +10,7 @@ import("//build/config/arm.gni")
 import("//build/config/loongarch64.gni")
 import("//build/config/mips.gni")
 import("//build_overrides/build.gni")
+import("//neva/neva.gni")
 
 declare_args() {
   libyuv_include_tests = !build_with_chromium
@@ -24,7 +25,8 @@ declare_args() {
   # errors on Fuchsia, macOS, and compilation errors on Windows.
   # TODO: bug 359006069 - Remove the restriction after the linker and
   # compilation errors are fixed.
-  libyuv_use_sme = current_cpu == "arm64" && (is_linux || is_android)
+  # TODO(neva): Remove !is_webos when webOS supports SME instructions.
+  libyuv_use_sme = current_cpu == "arm64" && (is_linux || is_android) && !is_webos
   libyuv_use_msa =
       (current_cpu == "mips64el" || current_cpu == "mipsel") && mips_use_msa
   libyuv_use_mmi =
-- 
2.34.1

